### PR TITLE
Sgratzl/heatmapfocus

### LINKED ISSUE #312: "Zoom to cluster" in heatmap view
The heatmap currently enables collapsing a clusters. Zoom to cluster, where only that cluster is visible, would also be useful. There would need to be a way to zoom back out to the full data.

Unsure how to work the UI such that both collapse and zoom are possible, would we need an interaction mode toggle?

### PR DESCRIPTION
**Pull request by sgratzl**
based on #308 

closes #312 

so far when the user clicks while holding the ctrl-button, it will focus on the node instead of collapsing it. @jeffbaumes what do you think?

image

### CODE REVIEW

**Comment by jeffbaumes** [comment]
On my mac, Ctrl-click brings up a the browser context menu, and the event does not seem to get to the page to trigger the zoom.

**Comment by sgratzl** [comment]
re mopdifier

> On my mac, Ctrl-click brings up a the browser context menu, and the event does not seem to get to the page to trigger the zoom.

re  modifiers, mac behaves differently. 

 * https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/ctrlKey
 * https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/altKey
 * https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/metaKey
 * https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/shiftKey

for now I also listen for `shift+click` events

File: web/src/components/vis/Heatmap.vue [hunk 1]
```diff
@@ -220,9 +237,15 @@ export default {
...
-      inner.select('text').html(d => (collapsed.has(d.data) ? MDI_PLUS_CIRCLE : MDI_MINUS_CIRCLE));
+      inner.select('text').html((d) => {
+        if (wrapper.focus === d.data) {
+          return MDI_STAR_CIRCLE;
+        }
+        return collapsed.has(d.data) ? MDI_PLUS_CIRCLE : MDI_MINUS_CIRCLE;
+      });
       inner.select('title').text(d => d.data.name);
       inner.classed('collapsed', d => collapsed.has(d.data));
+      inner.classed('focussed', d => wrapper.focus === d.data);
```
Comment (jeffbaumes) [inline web/src/components/vis/Heatmap.vue]: `focussed` -> `focused`

Both are acceptable, but the latter is much more common in American English.

https://english.stackexchange.com/questions/4791/focussed-or-focused-rules-for-doubling-the-last-consonant-when-adding-ed

File: web/src/components/vis/Heatmap.vue [hunk 2]
```diff
@@ -445,7 +468,8 @@ export default {
   fill: orange;
 }
 
-.nodes >>> .collapsed {
+.nodes >>> .collapsed,
+.nodes >>> .focussed {
```
Comment (jeffbaumes) [inline web/src/components/vis/Heatmap.vue]: Same.

**Pull request merged**